scipy/optimize/tests/test_slsqp.py

```diff
@@ -646,49 +708,50 @@ def f(x):
         with pytest.warns(RuntimeWarning, match='x were outside bounds'):
             res = minimize(f, x0, method='SLSQP', bounds=bounds)
             assert res.success
+            _check_kkt(res, bounds=bounds)
 
     def test_kkt_equality(self):
-        # an equality constraint mentioned in gh9839
+        # An equality constraint mentioned in gh9839
+        # Add a bound on a second, independent variable to test indexing
         def fun(x):
             return np.sum(x ** 2)
 
         def con_fun(x):
             return x[0] - 1
 
-        con = {'fun': con_fun, 'type': 'eq'}
-        res = minimize(fun, 3.0, constraints=[con], method='SLSQP')
+        cons = [{'fun': con_fun, 'type': 'eq'}]
+        bnds = [(None, None), (1.0, None)]
+        x0 = [3.0, 4.0]
+        res = minimize(fun, x0, method='SLSQP', constraints=cons, bounds=bnds)
+
         assert_allclose(res.kkt['eq'][0], np.array([2.0]))
+        _check_kkt(res, cons, bnds)
 
     def test_kkt_inequality(self):
-        # Test kkt multiplier return with example from GH14394. These are linear
-        # inequality constraints that can be specified either as constraints or
-        # bounds, so we can test both.
-        # To test if dimensions indices of bounds are correctly extracted, adds
-        # a second independent variable to the example from GH14394.
+        """Test kkt multiplier return with example from GH14394. These are
+        linear inequality constraints that can be specified either as
+        constraints or bounds, so we can test both. To test if dimensions
+        indices of bounds are correctly extracted, adds a second independent
+        variable to the example from GH14394."""
+
+        # The main constraint to be tested
         def con_fun1(x):
             return np.array([x[0] - 1.0, 2.0 - x[0]])
 
-        def con_jac1(x):
-            return np.array([[1.0, 0.0], [-1.0, 0.0]])
-
+        # Lower bound on x[1]
         def con_fun2(x):
-            return np.array([x[1] - 0.75])
-
-        def con_jac2(x):
-            return np.array([[0.0, 1.0]])
+            return x[1:] - 0.75
 
-        constraints = [
-            {'type': 'ineq', 'fun': con_fun1, 'jac': con_jac1},
-            {'type': 'ineq', 'fun': con_fun2, 'jac': con_jac2}
-        ]
+        cons = [{'type': 'ineq', 'fun': con_fun1},
+                {'type': 'ineq', 'fun': con_fun2}]
 
         # x[0] - 1 >=0 and 2 - x[0] >= 0. Equivalently, 1 <= x[0] <= 2.
-        bounds = [(1.0, 2.0), (0.75, np.inf)]
+        bnds = [(1.0, 2.0), (0.75, None)]
 
         x0 = np.array([1.5, 1.5])
 
         # Test cases for c < 1, 1 < c < 2, and c > 2
-        for c in [0.7, 1.5, 2.3]:
+        for c in [0.6, 1.5, 2.3]:
```

Comment:
I'm seeing a lot of changes that are not obviously necessary. Why wasn't 0.7 OK? Could you comment on all of these?

Reply:
Sorry - bad habits. I noticed my previous commit with the KKT multipliers for bounds was broken when I was playing with the tests, so I changed a lot of small things while trying to fix that.

For this change in particular, the values of 0.7 and 2.3 were the same distance from their respective bounds, so I just wanted to make sure we weren't getting lucky with some coincidental values.